src/main/java/cz/klemsa/tkbrew/controller/IngredientController.java

```diff
@@ -0,0 +1,220 @@
+package cz.klemsa.tkbrew.controller;
+
+import cz.klemsa.tkbrew.model.ingredient.*;
+import cz.klemsa.tkbrew.repository.ingredient.*;
+import org.bson.types.ObjectId;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Ingredient controller for CRUD operations.
+ * @author tomasklemsa
+ */
+@RestController
+@RequestMapping("/ingredient")
+public class IngredientController {
+    private static final String ID_PATH_VARIABLE = "{id:[0-9a-fA-F]{24}}"; // hex value
+
+    @Autowired
+    private IngredientRepository ingredientRepository;
+    @Autowired
+    private Logger log;
+
+    // --------------------------------------------------
+    // - GET Methods
+    // --------------------------------------------------
+
+    /**
+     * Gets ingredient by specific id.
+     * @param id in hexadecimal form like "5f7ac903e3ba595a5580e764"
+     * @return ingredient
+     */
+    @GetMapping("/" + ID_PATH_VARIABLE)
+    public ResponseEntity<Ingredient> getIngredient(@PathVariable("id") String id) {
+        if (ObjectId.isValid(id)) {
+            Optional<Ingredient>  ingredient = ingredientRepository.findById(new ObjectId(id));
+
+            if (ingredient.isEmpty()) {
+                log.info("Ingredient with id=" + id + " is not found.");
```

Comment:
Doporučuju používat zápis:
log.info("Ingredient with id={} is not found.", id);
je to rychlejší a čitelnější.